plugins/snapshot/local_snapshot.go

```diff
@@ -453,6 +452,7 @@ func createLocalSnapshotWithoutLocking(targetIndex milestone.Index, filePath str
 	if writeToDatabase {
 		// This has to be done before acquiring the SolidEntryPoints Lock, otherwise there is a race condition with "solidifyMilestone"
 		// In "solidifyMilestone" the LedgerLock is acquired, but by traversing the tangle, the SolidEntryPoint Lock is also acquired.
+		// ToDo: we should flush the caches here, just to be sure that all information before this local snapshot we stored in the persistence layer.
```

Comment:
future muxxer will  handle that? :carlos:

Reply:
Thats for future hornet team.... obviously thats not easy at all and only a really rare edge case. Just wanted to comment it here so we do know about that.

Reply:
isn't it just a call to `Tangle.Flush`?

I guess not, because then it would be easy so what am I misunderstanding?

Reply:
We can't call the flush method if other parts of the node still access the objects. Object storage is not yet capable of doing this. I already talked to hans, he wants to change something and implement a "timeheap", and as a side effect this will also be possible hopefully. I added that comment that we don't forget that in the future.